portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,22 +36,37 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-
     @Override
   public void doGet (HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     
     List<Comment> comments = new ArrayList<>();
+
+    int counter = 0;
     for (Entity entity : results.asIterable()) {
       long id = entity.getKey().getId();
+      String username = (String) entity.getProperty("username");
       String text = (String) entity.getProperty("text");
       String time = (String) entity.getProperty("time");
-      Comment comment = new Comment(id, text, time);
+      Comment comment = new Comment(id, username, text, time);
       comments.add(comment);
-     
+
+      counter++;
+      if (counter == 0) break;
+
+          
+      
     }
+     int showAmt = geShownComments(request);
```

Comment:
What if comments is empty? Can you return early here?

Reply:
good point, important edge case!

Reply:
good point, I was sort of thinking that this edge case was covered by the previous line of code:
if (counter == 0) break;

Reply:
actually, I think your right if the user requests 0 comments it should return early, I'll fix that